### ISSUE TITLE
Validation error doesn't improve during training

### ISSUE THREAD

**Issue by zain-sohail** [joppli/XRD_denoising#1]
After downloading the datasets from zenodo and trying to train the models (both VDSR and IRUNet), the validation error seems to barely improve.
For IRUNET:

```
Epoch 00041: val_loss did not improve from 0.53506
410/410 - 23s - loss: 0.5371 - mae: 0.2116 - val_loss: 0.5374 - val_mae: 0.2080 - lr: 5.0000e-04 - 23s/epoch - 55ms/step
Epoch 42/100

Epoch 00042: val_loss did not improve from 0.53506
410/410 - 23s - loss: 0.5386 - mae: 0.2129 - val_loss: 0.5362 - val_mae: 0.2067 - lr: 5.0000e-04 - 23s/epoch - 55ms/step
Epoch 43/100

Epoch 00043: val_loss did not improve from 0.53506
410/410 - 23s - loss: 0.5378 - mae: 0.2121 - val_loss: 0.5390 - val_mae: 0.2098 - lr: 5.0000e-04 - 23s/epoch - 56ms/step
Epoch 44/100

Epoch 00044: val_loss did not improve from 0.53506
410/410 - 23s - loss: 0.5378 - mae: 0.2123 - val_loss: 0.5390 - val_mae: 0.2092 - lr: 5.0000e-04 - 23s/epoch - 55ms/step
Epoch 45/100

Epoch 00045: val_loss improved from 0.53506 to 0.53492, saving model to output/trained_model/ckpt_loss=0.53492188.hdf5
410/410 - 23s - loss: 0.5379 - mae: 0.2127 - val_loss: 0.5349 - val_mae: 0.2054 - lr: 5.0000e-04 - 23s/epoch - 55ms/step
Epoch 46/100

Epoch 00046: val_loss did not improve from 0.53492
410/410 - 23s - loss: 0.5355 - mae: 0.2097 - val_loss: 0.5383 - val_mae: 0.2086 - lr: 5.0000e-04 - 23s/epoch - 55ms/step
Epoch 47/100

Epoch 00047: val_loss improved from 0.53492 to 0.53479, saving model to output/trained_model/ckpt_loss=0.53478527.hdf5
410/410 - 23s - loss: 0.5375 - mae: 0.2117 - val_loss: 0.5348 - val_mae: 0.2049 - lr: 5.0000e-04 - 23s/epoch - 55ms/step
Epoch 48/100

Epoch 00048: val_loss did not improve from 0.53479
410/410 - 23s - loss: 0.5367 - mae: 0.2111 - val_loss: 0.5358 - val_mae: 0.2060 - lr: 5.0000e-04 - 23s/epoch - 55ms/step
Epoch 49/100
```

I've tried with VDSR and it the loss stays at infinity.

Maybe not relevant but one thing I noticed in the dataset was that the image size described in journal (194 × 242 pixels) is actually (194 × 240 pixels I recall) in the dataset. 

Sidenote: Do you perhaps have the trained model available somewhere?

**Comment by joppli** [comment]
When training the models, make sure that you are using the appropriate tensorflow version (2.4.1). The optimization of the weights and biases for IRUNet is "smoother" compared to VDSR, which might explain why the loss of VDSR stays at infinity (see loss curves shown in Fig. 3(a) in the supplementary information of the paper). If you cannot use tensorflow 2.4.1 then it might also help to reduce the learning rate, and/or changing from MAE+MS-SSIM loss to the simpler MAE+SSIM loss. 

The trained model weights have now been uploaded to Zenodo (using images of size 192x240 pixels, therefore minor image pre-processing is required when using the Zenodo datasets).

**Comment by zain-sohail** [comment]
Could you comment on why the network would fail to train on a newer tensorflow version? I am using 2.11.

But I will try the suggested approaches.

**Comment by zain-sohail** [comment]
It seems the optimizer has changed in newer versions. Using tf.keras.optimizers.legacy.Adam allows the training to work
